pkg/odo/cli/pipelines/bootstrap.go

```diff
@@ -77,9 +103,127 @@ func (io *BootstrapParameters) Complete(name string, cmd *cobra.Command, args []
 		identifier := factory.NewDriverIdentifier(factory.Mapping(host, io.PrivateRepoDriver))
 		factory.DefaultIdentifier = identifier
 	}
+
+	// ask for sealed secrets only when default is absent
+	flagset := cmd.Flags()
+	if flagset.NFlag() == 0 {
+		err := initiateInteractiveMode(io)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := nonInteractiveMode(io)
+		if err != nil {
+			return err
+		}
+	}
+
+	io.Overwrite = true
+
+	return nil
+}
+
+// nonInteractiveMode gets triggered if a flag is passed, checks for mandatory flags.
+func nonInteractiveMode(io *BootstrapParameters) error {
+	mandatoryFlags := map[string]string{io.ServiceRepoURL: "service-repo-url", io.GitOpsRepoURL: "gitops-repo-url", io.ImageRepo: "image-repo"}
+	for key, value := range mandatoryFlags {
+		if key == "" {
+			return fmt.Errorf("The mandatory flag %q has not been set", value)
```

Comment:
```suggestion
			return fmt.Errorf("required flag(s) %q not set", value)
```
Just to keep the messages consistent between commands

Reply:
One last comment, we should probably keep the error messages consistent. Check the errors in other commands. Using `multierror` we can throw a single error for all the missing flags